### DISCUSSION TITLE
[Feature]: Switch to Ruff from black & flake8

### DISCUSSION THREAD

**Discussion by oscarmcm** [django-wiki/django-wiki#1317] [Ideas]
### I've already checked other feature proposals

Yes

### Is your feature request related to a problem?

No

### Describe your feature request.

Recently `Ruff` launched [their new format tool](https://astral.sh/blog/the-ruff-formatter), I believe that this is the last step needed to switch from `black` `flake8`

We might also need to update the `pre-commit` config to use `ruff`
https://github.com/django-wiki/django-wiki/blob/main/.pre-commit-config.yaml#L5

and as well, update the `pyproject.toml` where needed.

**Comment by oscarmcm** [comment]
tagging here @benjaoming to discuss it, and if so, move it to an issue.

**Reply by benjaoming** [reply]
I'm more than fine with this! But would be nice that it comes with a minimal set of changes to the codebase. But supposedly, Ruff is compatible with Black and Flake 8 so we shouldn't have issues.

**Comment by benjaoming** [comment]
Fixed by @oscarmcm :tada: https://github.com/django-wiki/django-wiki/pull/1321
